pennylane_lightning/core/src/utils/BLASLibLoaderManager.hpp

```diff
@@ -0,0 +1,92 @@
+// Copyright 2018-2024 Xanadu Quantum Technologies Inc.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file
+ * BLAS Lib dynamic loader manager.
+ */
+#pragma once
+
+#include <cstdlib>
+#include <filesystem>
+#include <iostream>
+#include <memory>
+#include <string>
+
+#ifndef SCIPY_OPENBLAS32_LIB
+#define SCIPY_OPENBLAS32_LIB ""
+#endif
+
+#include "SharedLibLoader.hpp"
+
+namespace Pennylane::Util {
+/**
+ * @brief BLAS Lib dynamic loader manager.
+ *
+ * This class is a singleton that manages the dynamic loading of BLAS libraries.
+ * It will search for the BLAS libraries in the given path, or in the RPATH
+ * - The path provided by the SCIPY_OPENBLAS32_LIB macro.
+ */
+class BLASLibLoaderManager {
```

Comment:
Do we expect this to be extensible? If not, lets declare it `final` if so, let's set the destructor as `virtual`